### PR TITLE
Enable experimental callflow feature for jenkins plugin

### PR DESCRIPTION
**Pull request by chriswininger**
Sonatype would like to begin more widely dog fooding the new callflow feature as well as allow customers to try it out. It is already available and dcoumented as an experimental feature in the IQ CLI, but has not yet been made available via the Jenkins Plugin.

This exposes callflow as a new experimental feature in the jenkins plugin. It can be enabled through optional arguments passed to nexusPolicyEvaluation.

for example:

```
nexusPolicyEvaluation(
                    iqApplication: 'test-callflow-3-11-2024',
                    iqInstanceId: 'localhost-iq-server',
                    iqStage: 'build',
                    iqScanPatterns: [[scanPattern: 'test-callflow/**/*.jar']],
                    runCallflow: true,
                    callflowConfiguration: [
                       callflowScanPatterns: [[scanPattern: 'test-callflow/**/*with-dependencies.jar']],
                       callflowNamespaces: ['foo.com'],
                       additionalConfiguration: [foo: "bar", test: 1]
                   ]
)
```

`runCallflow` defaults to false

`callflowConfiguration` is optional. When not provided callflow will be invoked using the scanTargets that are produced from the iqScanPatterns.

Each of the parameters under callflowConfiguration are optional. `callflowScanPatterns` lets you set an alternative set of patterns to match on for callflow. `callflowNamespaces` let's you send a list of name spaces for callflow to use when limiting how it does it's analysis. `additionalConfiguration` is a map of String to Object which allows for passing in undocumented experimental configuration to callflow. This will primarily be used internally or when troubleshooting.

*Passing Build:* https://jenkins.ci.sonatype.dev/job/integrations/job/jenkins/job/feature-snapshots/job/SDEV-941-Enable-Callflow-For-Jenkins-Plugin/6/

*Jira Ticket:* https://sonatype.atlassian.net/browse/SDEV-940

### Testing done

I have built the plugin on local alongside the new version of the nexus-java-api. I have deployed the plugin to a locally running jenkins docker container and tested against my changes in several scenarios:

1. with callflow disabled
2. with callflow enabled and all default options
3. with callflow enabled and various combinations of callflowConfiguration options

```[tasklist]
### Submitter checklist
- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- [X] Link to relevant pull requests, esp. upstream and downstream changes (Downstream changes needed are merged) 
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -88,7 +88,8 @@
     <enforcer.skip>true</enforcer.skip> <!-- TODO numerous requireUpperBoundDeps, some probably indicative of real problems -->
     <jvnet-localizer-plugin.version>1.23</jvnet-localizer-plugin.version>
     <forkCount>1</forkCount>
-    <nexus-platform-api.version>5.0.3-01</nexus-platform-api.version>
+
+    <nexus-platform-api.version>5.0.4-SNAPSHOT</nexus-platform-api.version>
```
Comment (chriswininger) [inline pom.xml:92]: SNAPSHOT will be removed once we have released the nexus-platform API

File: src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorUtil.groovy [hunk 2]
```diff
@@ -110,7 +112,32 @@ class IqPolicyEvaluatorUtil
...
 
         File workDirectory = new File(workspace.getRemote())
-        evaluationResult = iqClient.evaluateApplication(applicationId, iqStage, scanResult, workDirectory)
+
+        final CallflowOptions callflowOptions
+
+        if (iqPolicyEvaluator.getRunCallflow()) {
+          listener.logger.println('callflow analysis is enabled')
+
+          CallflowConfiguration callflowConfiguration = iqPolicyEvaluator.getCallflowConfiguration()
```
Comment (jpizaST) [inline src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorUtil.groovy:121]: I was looking at the code but I did not find where the `CallFlowConfiguration` is being instantiated/injected/set into the `iqPolicyEvaluator`. How does this work?

**Review by jpizaST — APPROVED** [review]

File: src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorUtil.groovy (hunk 2 above)
Comment (chriswininger) [reply src/main/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorUtil.groovy:121]: This is handled by the plugin framework. That's why there are those DataBoundConstructor annotations

**Review by eduard-tita — APPROVED** [review]
+1

**Pull request merged**